sql/src/test/java/org/apache/druid/sql/calcite/util/SqlTestFramework.java

```diff
@@ -852,11 +862,16 @@ public PlannerFixture(
           new DruidHookDispatcher()
       );
       componentSupplier.finalizePlanner(this);
-      this.statementFactory = QueryFrameworkUtils.createSqlStatementFactory(
+      final SqlToolbox toolbox = new SqlToolbox(
           framework.engine,
           plannerFactory,
-          authConfig
+          new ServiceEmitter("dummy", "dummy", new NoopEmitter()),
+          new NoopRequestLogger(),
+          QueryStackTests.DEFAULT_NOOP_SCHEDULER,
+          new DefaultQueryConfig(ImmutableMap.of()),
+          new SqlLifecycleManager()
```

Comment:
its better to use the injector to obtain the right instances ;  it should already have almost all of this...I've seen issue from like a random empty query config was picked up somewhere and made things go south...

Reply:
You want to bind the statement factory with like a different version of `SqlStatementFactoryModule` that makes this instead of the regular one I guess instead of making it here? I guess can do that, will look into it later